libnmstate/nm/wired.py

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate import nmclient
+
+
+def get_info(active_connection):
+    """
+    Provides the current active values for an active connection.
+    """
+    info = {}
+    # FIXME: Maybe use
+    # active_connection.get_devices()[0].get_applied_connection(0) to get the
+    # real runtime connection...
+    try:
+        remote_connection = active_connection.get_connection()
+        wired_config = remote_connection.get_setting(nmclient.NM.SettingWired)
```

Comment:
Why not use get_setting_wired() ?

Reply:
I hope to reuse the code and make it more generic when adding more settings. Then only the argument to `get_setting()` is generic, which is a bit nicer that to do something like `getattr(remote_connection, 'get'setting_wired')()`.

Reply:
I do not see how you will make it more generic this way, I only see here the need to depend on nmclient.NM.* (and require a separate mock for it), especially when for each of these you need to check specific attributes.
But I guess we will see it when you add another property.